app/workers/backend_delete_service_worker.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class BackendDeleteServiceWorker
+  include Sidekiq::Worker
+
+  def self.enqueue(event)
+    perform_async(event.event_id)
+  end
+
+  def perform(event_id)
+    event = EventStore::Repository.find_event!(event_id)
+    service_id = event.service_id
+    batch = Sidekiq::Batch.new
+    batch.description = "Deleting in Backend the Service ##{service_id}"
+    batch.on(:success, self.class, {'service_id' => service_id})
+    batch.jobs do
+      BackendDeleteEndUsersWorker.perform_async(event_id)
+      BackendDeleteStatsWorker.perform_async(event_id)
+    end
+  end
+
+  def on_success(_bid, options)
```

Comment:
or `on_complete` ?

Reply:
@macejmic If we make `on_complete` means that even if deleting stats didn't return a `200` from Backend, we carry on deleting the Service from Backend, and in that case the stats for that Service will never be deleted 😞

Reply:
makes sense.